tests/e2e/configurer/chain/commands.go

```diff
@@ -38,21 +43,38 @@ type params struct {
 func (n *NodeConfig) CreateBalancerPool(poolFile, from string) uint64 {
 	n.LogActionF("creating balancer pool from file %s", poolFile)
 	cmd := []string{"osmosisd", "tx", "gamm", "create-pool", fmt.Sprintf("--pool-file=/osmosis/%s", poolFile), fmt.Sprintf("--from=%s", from)}
-	_, _, err := n.containerManager.ExecTxCmd(n.t, n.chainId, n.Name, cmd)
+	resp, _, err := n.containerManager.ExecTxCmd(n.t, n.chainId, n.Name, cmd)
 	require.NoError(n.t, err)
 
-	poolID := n.QueryNumPools()
+	// TODO: create a helper function for parsing pool ID and prop ID from the response
```

Comment:
This and the other similar value extractions from the response could be done generically. Gjson (which we're using on other tests) may help here.

What's the reason to fetch from the response instead of querying though?

Reply:
In this instance, querying num pools is not always valid. For instance, one of the parallel tests could create a pool in between this test creating a pool and querying num pools. So pulling direct from the response guarantees corectness.